components/server/src/billing/entitlement-service-ubp.ts

```diff
@@ -63,69 +62,63 @@ export class EntitlementServiceUBP implements EntitlementService {
         if (result.reached) {
             return result.attributionId;
         }
         return undefined;
     }
 
-    private async getMaxParallelWorkspaces(user: User, date: Date): Promise<number> {
-        if (await this.hasPaidSubscription(user, date)) {
+    private async getMaxParallelWorkspaces(userId: string, organizationId: string): Promise<number> {
+        if (await this.hasPaidSubscription(userId, organizationId)) {
             return MAX_PARALLEL_WORKSPACES_PAID;
         } else {
             return MAX_PARALLEL_WORKSPACES_FREE;
         }
     }
 
-    async maySetTimeout(user: User, date: Date): Promise<boolean> {
-        return this.hasPaidSubscription(user, date);
+    async maySetTimeout(userId: string, organizationId?: string): Promise<boolean> {
+        return this.hasPaidSubscription(userId, organizationId);
     }
 
-    async getDefaultWorkspaceTimeout(user: User, date: Date): Promise<WorkspaceTimeoutDuration> {
-        if (await this.hasPaidSubscription(user, date)) {
+    async getDefaultWorkspaceTimeout(userId: string, organizationId: string): Promise<WorkspaceTimeoutDuration> {
+        if (await this.hasPaidSubscription(userId, organizationId)) {
             return WORKSPACE_TIMEOUT_DEFAULT_LONG;
         } else {
             return WORKSPACE_TIMEOUT_DEFAULT_SHORT;
         }
     }
 
-    async getDefaultWorkspaceLifetime(user: User, date: Date): Promise<WorkspaceTimeoutDuration> {
-        if (await this.hasPaidSubscription(user, date)) {
+    async getDefaultWorkspaceLifetime(userId: string, organizationId: string): Promise<WorkspaceTimeoutDuration> {
+        if (await this.hasPaidSubscription(userId, organizationId)) {
             return WORKSPACE_LIFETIME_LONG;
         } else {
             return WORKSPACE_LIFETIME_SHORT;
         }
     }
 
-    /**
-     * DEPRECATED: With usage-based billing, users can choose exactly how many resources they want to get.
-     * Thus, we no longer need to "force" extra resources via the `userGetsMoreResources` mechanism.
-     */
-    async userGetsMoreResources(user: User, date: Date = new Date()): Promise<boolean> {
-        return false;
-    }
-
     /**
      * Returns true if network connections should be limited
      * @param user
      */
-    async limitNetworkConnections(user: User, date: Date): Promise<boolean> {
+    async limitNetworkConnections(userId: string, organizationId: string): Promise<boolean> {
         // gpl: Because with the current payment handling (pay-after-use) having a "paid" plan is not a good enough classifier for trushworthyness atm.
         // We're looking into improving this, but for the meantime we limit network connections for everybody to reduce the impact of abuse.
         return true;
     }
 
-    private async hasPaidSubscription(user: User, date: Date): Promise<boolean> {
+    private async hasPaidSubscription(userId: string, organizationId?: string): Promise<boolean> {
+        if (organizationId) {
```

Comment:
This is good and we should even aim for making the organization mandatory, but it is a breaking change. For instance, users who were able to change timeouts on workspaces running on their free org (but have another paid org) can now no longer do this.

Reply:
Exactly. But sharpening that behavior is sth we should have done all along, right after the migration to Stripe.